plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -798,13 +907,20 @@ private void writeEqualityDeleteToNationTable(Table icebergTable, Optional<Parti
 
     private void writeEqualityDeleteToNationTable(Table icebergTable, Optional<PartitionSpec> partitionSpec, Optional<PartitionData> partitionData, Map<String, Object> overwriteValues)
             throws Exception
+    {
+        writeEqualityDeleteToNationTableWithDeleteColumns(icebergTable, partitionSpec, partitionData, overwriteValues, Optional.empty());
+    }
+
+    private void writeEqualityDeleteToNationTableWithDeleteColumns(Table icebergTable, Optional<PartitionSpec> partitionSpec, Optional<PartitionData> partitionData, Map<String, Object> overwriteValues, Optional<List<String>> deleteFileColumns)
+            throws Exception
     {
         Path metadataDir = new Path(metastoreDir.toURI());
         String deleteFileName = "delete_file_" + UUID.randomUUID();
         FileIO fileIo = new ForwardingFileIo(fileSystemFactory.create(SESSION));
 
-        Schema deleteRowSchema = icebergTable.schema().select(overwriteValues.keySet());
-        List<Integer> equalityFieldIds = overwriteValues.keySet().stream()
+        List<String> deleteColumns = deleteFileColumns.orElse(new ArrayList<>(overwriteValues.keySet()));
+        Schema deleteRowSchema = icebergTable.schema().select(deleteColumns);
```

Comment:
This method of selecting the schema doesn't preserve the order in which `deleteColumns` were provided.
Alternative method:

```
        Schema deleteRowSchema = new Schema(deleteColumns.stream()
                .map(name -> icebergTable.schema().findField(name))
                .collect(toImmutableList()));

```